### ISSUE TITLE
Apply to caption task.

### ISSUE THREAD

**Issue by MarcusNerva** [jacobgil/pytorch-grad-cam#77]
Greetings!
Actually, I am studying how to utilize this visualization technology to analyze the process of video captioning.  Here I will briefly introduce the video captioning process, which is quite simple.

![image](https://user-images.githubusercontent.com/37070207/116123109-30bfb300-a6f5-11eb-9efe-53b19e9ad212.png)

As you can see, video captioning is actually a classification task. The number of categories is the size of the dictionary. 

Can I simply imitate the process of applying CAM on the image classification task, just backpropagate the grad and check which part of the frames caught the model's attention?

**Comment by jacobgil** [comment]
Yes.
At every time point t, you can compute GradCAM for each of the input frames (i <= t).

- You have the loss, which is just loss = p[s_t][target_category]
- You can call loss.backward()
- The target_layer should be the regular target layer in the CNN feature extractor, e.g if your custom model has a member called resnet, then `target_layer= video_caption_model.resnet.features[-1]`
- The gradients and activations at time point will probably be of size `t x 2048 x Height x Width`, since the gradient is computed for all t frames.

Hope this helps.

**Comment by jacobgil** [comment]
Oopsy, ignore the previous comment, I got it wrong.

**Comment by MarcusNerva** [comment]
> I editted the previous comment.
> 
> I'm trying to think if there is a way to simplify this..

Thank you for your advise. I will have a try.
